src/Lecture2.hs

```diff
@@ -97,7 +147,21 @@ spaces.
 
 🕯 HINT: look into Data.Char and Prelude modules for functions you may use.
 -}
-dropSpaces = error "TODO"
+
+dropSpaces :: [Char] -> [Char]
+dropSpaces  = leaveOnlyLetters . dropWhitespacesLeft
+   where
+      leaveOnlyLetters :: [Char] -> [Char]
+      leaveOnlyLetters s
+         | null s        = s
+         | head s /= ' ' = (head s) : (leaveOnlyLetters (tail s))
+         | otherwise     = []
+
+      dropWhitespacesLeft :: [Char] -> [Char]
+      dropWhitespacesLeft s
+         | null s        = s
+         | head s == ' ' = dropWhitespacesLeft (tail s)
+         | otherwise     = s
```

Comment:
Have a look at standard functions `dropWhile` and `takeWhile` 🙂 

You can use them and you won't need to implement your own recursive functions.